util/util.go

```diff
@@ -315,3 +315,12 @@ func (k *KeyServiceClient) MakeRequest(method string, path string, body interfac
 	}
 	return nil
 }
+
+func GetRequestLocale(explicitLocale string, r *http.Request) string {
+	if explicitLocale != "" {
+		return explicitLocale
+	}
+
+	// Use header as fallback
+	return r.Header.Get("Accept-Language")
```

Comment:
Let's add some validation here. Maybe something like `[a-zA-Z0-9\-]{2,20}`?

According to [MDN](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/languages), these language codes use the format defined in [RFC5646](https://datatracker.ietf.org/doc/html/rfc5646#appendix-A).

Reply:
Also, we could trim it, just in case.

Reply:
It's it's already validated elsewhere, then that sounds good to me.